velox/dwio/parquet/reader/PageReader.cpp

```diff
@@ -121,6 +123,104 @@ const char* PageReader::readBytes(int32_t size, BufferPtr& copy) {
   return copy->as<char>();
 }
 
+const char* FOLLY_NONNULL decompressLz4AndLzo(
+    const char* compressedData,
+    BufferPtr& uncompressedData,
+    uint32_t compressedSize,
+    uint32_t uncompressedSize,
+    memory::MemoryPool& pool,
+    const thrift::CompressionCodec::type codec_) {
+  dwio::common::ensureCapacity<char>(uncompressedData, uncompressedSize, &pool);
+
+  uint32_t decompressedTotalLength = 0;
+  auto* inputPtr = compressedData;
+  auto* outPtr = uncompressedData->asMutable<char>();
+  uint32_t inputLength = compressedSize;
+
+  while (inputLength > 0) {
+    if (inputLength < sizeof(uint32_t)) {
+      VELOX_FAIL(
+          "{} uncompress failed, input len is to small: {}",
+          codec_,
+          inputLength)
+    }
+    uint32_t uncompressedBlockLength =
+        __builtin_bswap32(velox::dwio::common::load32(inputPtr));
```

Comment:
`folly::Endian::big`

Reply:
Hi @Yuhta thank you for your reply, I have modified the code according to the suggestions.